### PR TITLE
h2: add max_pending_accept_reset_streams configuration option

### PR DESCRIPTION
**Pull request by Noah-Kennedy**
This allows users to set the configuration option from https://github.com/hyperium/h2/pull/668.

### CODE REVIEW

File: src/proto/h2/server.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ const DEFAULT_CONN_WINDOW: u32 = 1024 * 1024; // 1mb
 const DEFAULT_STREAM_WINDOW: u32 = 1024 * 1024; // 1mb
 const DEFAULT_MAX_FRAME_SIZE: u32 = 1024 * 16; // 16kb
 const DEFAULT_MAX_SEND_BUF_SIZE: usize = 1024 * 400; // 400kb
-// 16 MB "sane default" taken from golang http2
+                                                     // 16 MB "sane default" taken from golang http2
```
Comment (LPardue) [inline src/proto/h2/server.rs:38]: errm

Comment (Noah-Kennedy) [reply src/proto/h2/server.rs:38]: huh, fixing

Comment (Noah-Kennedy) [reply src/proto/h2/server.rs:38]: its rustfmt

Comment (Noah-Kennedy) [reply src/proto/h2/server.rs:38]: I can do nothing here, it is rustfmt

File: Cargo.toml [hunk 2]
```diff
@@ -13,10 +13,10 @@ categories = ["network-programming", "web-programming::http-client", "web-progra
...
 
 include = [
-  "Cargo.toml",
-  "LICENSE",
-  "src/**/*",
-  #"build.rs",
+    "Cargo.toml",
+    "LICENSE",
+    "src/**/*",
+    #"build.rs",
```
Comment (hawkw) [inline Cargo.toml:19]: nit: unrelated formatting change?
```suggestion
  "Cargo.toml",
  "LICENSE",
  "src/**/*",
  #"build.rs",
```

File: src/proto/h2/server.rs (hunk 1 above)
Comment (hawkw) [inline src/proto/h2/server.rs:38]:
```suggestion
// 16 MB "sane default" taken from golang http2
```

File: src/server/conn.rs [hunk 3]
```diff
@@ -394,6 +394,17 @@ impl<E> Http<E> {
         self
     }
 
+    /// Configures the maximum number of pending reset streams allowed before a GOAWAY will be sent.
+    ///
+    /// See <https://github.com/hyperium/hyper/issues/2877> for more information.
+    #[cfg(feature = "http2")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "http2")))]
+    pub fn http2_max_pending_accept_reset_streams(&mut self, max: usize) -> &mut Self {
```
[the remark this answers is not in the document]
Comment (hawkw) [reply src/server/conn.rs:402]: We should definitely mention the default value in the docs. I'm +1 on taking `impl Into<Option<usize>>` here, but :woman_shrugging:

File: src/proto/h2/server.rs (hunk 1 above)
Comment (Noah-Kennedy) [reply src/proto/h2/server.rs:38]: Rustfmt was doing this, but ill take this for the same of a small diff

File: src/server/conn.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (Noah-Kennedy) [reply src/server/conn.rs:402]: I have zero preference either way. I'll just go ahead and change it.

**Comment by seanmonstar** [comment]
Does it seem likely people want to configure this on the client? I'm guessing since hyper disables server push, there's nothing to worry about?

**Comment by Noah-Kennedy** [comment]
> Does it seem likely people want to configure this on the client? I'm guessing since hyper disables server push, there's nothing to worry about?

I don't think we need client configuration for now, if someone asks then we can add it IMO.

**Comment by Noah-Kennedy** [comment]
Like you said, we don't support server push anyways for the client.

File: src/server/conn.rs [hunk 4]
```diff
@@ -394,6 +394,22 @@ impl<E> Http<E> {
         self
     }
 
+    /// Configures the maximum number of pending reset streams allowed before a GOAWAY will be sent.
+    ///
+    /// This will default to whatever the default in h2 is. As of v0.3.17, it is 20.
```
Comment (hawkw) [inline src/server/conn.rs:399]: nit: is there documentation in `h2` we can link to for this?

Comment (hawkw) [reply src/server/conn.rs:399]: also, silly docs nit: I might say 
```suggestion
    /// This will default to the default value set by the [`h2` crate](https://crates.io/crates/h2).
    /// As of v0.3.17, it is 20.
```
or something, to make it obvious that when we say `h2` here, we are referring to the library?

**Review by hawkw — APPROVED** [review]
some last docs quibbles, but LGTM otherwise!

**Comment by Noah-Kennedy** [comment]
I committed your suggestions
